src/journal.rs

```diff
@@ -225,6 +226,38 @@ impl Journal {
         sd_try!(ffi::sd_journal_get_realtime_usec(self.j, &mut timestamp_us));
         Ok(system_time_from_realtime_usec(timestamp_us))
     }
+
+    /// Adds a match by which to filter the entries of the journal.
+    /// If a match is applied, only entries with this field set will be iterated.
+    pub fn match_add<T: Into<Vec<u8>>>(&self, key: &str, val: T) -> Result<&Journal> {
```

Comment:
I understand the concern. This is somehow similar to other rust types providing interior mutability, minus the explicit borrowing part. And this is something that doesn't concern only this new match subsystem: seek and next_record also alters internal cursor position.

I think that conveying this semantic via comment is neither efficient nor idiomatic. What about moving to explicit `&mut self`, even if not strictly required by rust, to let the caller be aware of this?